lib/registry/src/wasmer_env.rs

```diff
@@ -42,6 +42,22 @@ impl WasmerEnv {
         }
     }
 
+    pub fn registry_public_url(&self) -> Result<Url, Error> {
+        let mut url = self.registry_endpoint()?;
+        url.set_path("");
+
+        let domain = url
+            .host_str()
+            .context("url has no host")?
+            .strip_prefix("registry.")
+            .context("could not derive registry public url")?
+            .to_string();
+        url.set_host(Some(&domain))
+            .context("could not derive registry public url")?;
+
+        Ok(url)
+    }
```

Comment:
This feels a bit hacky. We should really be [asking the registry](https://registry.wasmer.io/graphql?query=query%20%7B%0A%20%20info%20%7B%0A%20%20%20%20defaultFrontend%0A%20%20%7D%0A%7D) for the frontend URL.

```graphql
query {
  info {
    defaultFrontend
  }
}
```